File: iosSDK/text/KVC/KVC/ViewController.m
//
//  ViewController.m
//  KVC
//
//  Created by  wj on 16/5/4.
//  Copyright © 2016年 gzq. All rights reserved.
//

#import "ViewController.h"
#import "Student.h"

@interface ViewController (){
    
    Student *student;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    student = [[Student alloc]init];
    student.name = @"韦德";
    
    //KVC -->KEY VALUE Coding
    [student setValue:@"白边" forKey:@"name"];
    NSLog(@"%@",student.name);
    
    
    //KVO -->KEY VALUE Observer
    [student addObserver:self forKeyPath:@"name" options:NSKeyValueObservingOptionNew  context:nil];
    
    student.name = @"德罗赞";
    NSLog(@"%@",student.name);
    
    // Do any additional setup after loading the view, typically from a nib.
}

-(void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary<NSString *,id> *)change context:(void *)context{
    
    NSString *old = [change objectForKey:@"old"];
    NSString *new = [change objectForKey:@"new"];
    
    NSLog(@"%@ %@",old,new);
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: oop/oop/ViewController.swift
//
//  ViewController.swift
//  oop
//
//  Created by  wj on 15/10/27.
//  Copyright © 2015年 gzq. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
var MiniDuck = Duck()
    override func viewDidLoad() {
        super.viewDidLoad()
        
        MiniDuck.flyBehavior.FlyWithWings()
        MiniDuck.quackBehavior.Squeak()

        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

